examples/chat2-reliable/peer_retrieval.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"chat2-reliable/pb"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"github.com/waku-org/go-waku/waku/v2/peermanager"
+	wpb "github.com/waku-org/go-waku/waku/v2/protocol/pb"
+	"github.com/waku-org/go-waku/waku/v2/protocol/relay"
+	"github.com/waku-org/go-waku/waku/v2/protocol/store"
+	"google.golang.org/protobuf/proto"
+)
+
+// below functions are specifically for peer retrieval of missing msgs instead of store
+func (c *Chat) doRequestMissingMessageFromPeers(messageID string) error {
+	peers := c.node.Host().Network().Peers()
+	for _, peerID := range peers {
```

Comment:
Any particula reason why you request the message from **every** known peer?

Reply:
If I understand the logic correctly it will stop requesting once the first peer responds. Should be a reasonable approach.